doc/docs.md

```diff
@@ -1331,6 +1331,45 @@ println(numbers) // [1, 2, 3]
 ```
 When an identifier is just a single underscore, it is ignored.
 
+##### Custom iterators
+Types that implement a `next` method returning an `Option` can be iterated
+with a `for` loop. 
+
+```v
+struct SquareIterator {
+	arr []int
+mut:
+	idx int
+}
+
+fn (mut iter SquareIterator) next() ?int {
+	if iter.idx >= iter.arr.len {
+		return error('')
```

Comment:
I think this should be `return none` instead of `return error('')` as it makes more sense for an iterator to return `none` when there are no more items.

It might make sense to reset `iter.idx = 0` here to allow multiple iterations as well.

Reply:
Is `none` going to stay a "thing" in V?

Reply:
I think so? At least in V's current state `none` would be the way to go, but I guess we don't want to change the docs too much..